process.c

```diff
@@ -1520,48 +1512,41 @@ rb_detach_process(rb_pid_t pid)
 
 /*
  *  call-seq:
- *     Process.detach(pid)   -> thread
+ *    Process.detach(pid) -> thread or nil
  *
- *  Some operating systems retain the status of terminated child
- *  processes until the parent collects that status (normally using
- *  some variant of <code>wait()</code>). If the parent never collects
- *  this status, the child stays around as a <em>zombie</em> process.
- *  Process::detach prevents this by setting up a separate Ruby thread
- *  whose sole job is to reap the status of the process _pid_ when it
- *  terminates. Use #detach only when you do not intend to explicitly
- *  wait for the child to terminate.
+ *  Avoids the potential for a child process to become a
+ *  {zombie process}[https://en.wikipedia.org/wiki/Zombie_process].
+ *  This method is needed only when the parent process will never wait
+ *  for the child process.
  *
- *  The waiting thread returns the exit status of the detached process
- *  when it terminates, so you can use Thread#join to
- *  know the result.  If specified _pid_ is not a valid child process
- *  ID, the thread returns +nil+ immediately.
```

Comment:
This doesn't look correct. The function `rb_detach_process` can never return `nil`, it always returns a new thread. It looks like the case in testing as well:

```ruby
Process.detach(213131212) # => #<Process::Waiter:0x00007f190ad24338 run>
```

Reply:
I don't know what needs to be changed here.

Reply:
I think the call-seq needs to be changed. It always returns a `thread` and never `nil`.

Reply:
Fixed.